src/components/forms/CharacterCount/CharacterCount.tsx

```diff
@@ -0,0 +1,165 @@
+import React, { useState } from 'react'
+import classnames from 'classnames'
+
+import { TextInput, TextInputRef } from '../TextInput/TextInput'
+import { Textarea, TextareaRef } from '../Textarea/Textarea'
+import { FormGroup } from '../FormGroup/FormGroup'
+
+interface CharacterCountProps {
+  id: string
+  name: string
+  maxLength: number
+  className?: string
+  label?: React.ReactNode
+  isTextArea?: boolean
+  textRef?: // currently getting an error when adding this as ref
+  | string
+    | ((instance: HTMLInputElement | null) => void)
+    | ((instance: HTMLTextAreaElement | null) => void)
+    | React.RefObject<HTMLInputElement>
+    | React.RefObject<HTMLTextAreaElement>
+    | null
+    | undefined
+}
+
+export const CharacterCount = (
+  props: CharacterCountProps &
+    React.InputHTMLAttributes<HTMLInputElement> &
+    React.TextareaHTMLAttributes<HTMLTextAreaElement>
+): React.ReactElement => {
+  const {
+    id,
+    name,
+    className,
+    label,
+    maxLength,
+    defaultValue,
+    isTextArea,
+    textRef,
+    type, // Conflicts with type attribute already set on TextInput component
+    ...inputProps
```

Comment:
I'm not sure it makes sense to lump the Input and TextArea attributes. I think they are distinct and should be applied to the textArea or Input respectively.

Reply:
I don't think it's too bad... the thing is we'd need to figure out some conditionally-required props, ie. require `textareaProps` if it's a textarea and `textInputProps` if not. I'm not sure if this is possible with TS but I think treating them as the same is okay, and TS checking should catch issues if you tried to pass input-only props to a textarea and vice-versa. An alternative could be to create separate `CharacterCountTextInput` and `CharacterCountTextarea` components 🤷‍♀️

Reply:
I am but an 🥚 but since they are need mutually exclusive props (and are separate html tags), I would vote for separate components.

Reply:
In my opinion I don't think the implementation is different enough to separate the components.  Textarea just happens to have some quirky attributes that textinput doesn't like rows, columns, wrap behavior, spellcheck ... USWDS seems to ignore rows and column attributes anyway at least within a default form group.  Honestly there are more `<input>` types besides `type="text"` to probably consider supporting or that would diverge more than textarea does.